### PR TITLE
Feature/implement wireguard dvpn client

### PR DESCRIPTION
**Pull request by yeszhanov95**
These changes allow use dVPN via Superproxy

### CODE REVIEW

File: services/wireguard/endpoint/dvpnclient/client.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (C) 2018 The "MysteriumNetwork/node" Authors.
```
Comment (n10ty) [inline services/wireguard/endpoint/dvpnclient/client.go:2]: 2022

File: services/wireguard/endpoint/dvpnclient/client.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+	err := c.configureDevice(config)
+	if err != nil {
+		rollback.Run()
+		return err
+	}
+
+	if config.Peer.Endpoint != nil {
+		gw := config.Subnet.IP.To4()
+		gw[3]--
```
Comment (n10ty) [inline services/wireguard/endpoint/dvpnclient/client.go:74]: This is potential panic.
```
If ip is not an IPv4 address, To4 returns nil.
```

File: services/wireguard/endpoint/dvpnclient/client.go [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+		if err := cmdutil.SudoExec("ip", "route", "add", "default", "via", gw.String(), "dev", config.IfaceName, "table", strings.TrimLeft(config.IfaceName, "myst")); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *client) ReConfigureDevice(config wgcfg.DeviceConfig) error {
+	err := c.configureDevice(config)
```
Comment (n10ty) [inline services/wireguard/endpoint/dvpnclient/client.go:84]:
```go
return c.configureDevice(config)
```

File: cmd/di.go [hunk 4]
```diff
@@ -948,7 +948,7 @@ func (di *Dependencies) bootstrapLocationComponents(options node.Options) (err e
 
 	di.LocationResolver = location.NewCache(resolver, di.EventBus, time.Minute*5)
 
-	if !config.GetBool(config.FlagProxyMode) {
+	if !config.GetBool(config.FlagProxyMode) && !config.GetBool(config.FlagDVPNMode) {
```
Comment (soffokl) [inline cmd/di.go:951]: Should it be `||`?

File: config/flags_node.go [hunk 5]
```diff
@@ -214,6 +214,13 @@ var (
 		Value: false,
 	}
 
+	// FlagProxyMode allows running node in a kernelspace for dVPN via Superproxy.
+	FlagDVPNMode = cli.BoolFlag{
+		Name:  "dvpnmode",
+		Usage: "Run in a kernelspace for dVPN via Superproxy",
```
Comment (soffokl) [inline config/flags_node.go:220]:
```suggestion
		Usage: "Run in a kernelspace without establishing system-wite tunnels",
```

**Review by soffokl — APPROVED** [review]

File: cmd/di.go (hunk 4 above)
Comment (yeszhanov95) [reply cmd/di.go:951]: we subscribe to that topic only if node running not in `proxymode` AND not in `dvpnmode`. that's why it should be `&&`

**Pull request merged**